### PR TITLE
Implemented handling of server side locations_checked for location helper

### PR DESCRIPTION
**Pull request by JarnoWesthof**

### CODE REVIEW

File: Archipelago.MultiClient.Net.Tests/LocationCheckHelperFixture.cs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+            Assert.Contains(3, sut.AllLocations);
+
+            Assert.Contains(1, sut.AllLocationsChecked.ToArray());
+            Assert.Contains(3, sut.AllLocationsChecked.ToArray());
+
+            Assert.Contains(2, sut.AllMissingLocations);
+        }
+
+        [Test]
+        public void Enumeratrion_over_collection_should_not_throw_when_new_data_is_received()
```
Comment (Ijwu) [inline Archipelago.MultiClient.Net.Tests/LocationCheckHelperFixture.cs:150]:
```suggestion
        public void Enumeration_over_collection_should_not_throw_when_new_data_is_received()

```

File: Archipelago.MultiClient.Net/Packets/RoomUpdatePacket.cs [hunk 2]
```diff
@@ -13,8 +13,5 @@ public class RoomUpdatePacket : RoomInfoPacket
 
         [JsonProperty("checked_locations")]
         public List<int> CheckedLocations { get; set; }
-
-        [JsonProperty("missing_locations")]
-        public List<int> MissingLocations { get; set; }
```
Comment (Ijwu) [inline Archipelago.MultiClient.Net/Packets/RoomUpdatePacket.cs:18]: Why remove this property? It is part of the RoomUpdatePacket as far as I can tell.

Comment (JarnoWesthof) [reply Archipelago.MultiClient.Net/Packets/RoomUpdatePacket.cs:18]: No actually its not, 
https://github.com/ArchipelagoMW/Archipelago/blob/main/docs/network%20protocol.md#arguments-5
states "Should never be sent as an update, if needed is the inverse of checked_locations.", which basicly means you wont find it on an RoomUpdatedPacket ever

Wich is also confirmed here: https://discord.com/channels/731205301247803413/731214280439103580/926991639912788018

**Comment by JarnoWesthof** [comment]
Fixes #23

**Comment by JarnoWesthof** [comment]
Has only been unit tested, planning to implement this into Timespinner later today to see how it behaves in a real client implementation, maybe the initial checked_locations from the connected packet should also trigger the event handler but that still up for debate

**Comment by JarnoWesthof** [comment]
Tested with timespinner seems to work fine

File: Archipelago.MultiClient.Net/Packets/RoomUpdatePacket.cs (hunk 2 above)
Comment (Ijwu) [reply Archipelago.MultiClient.Net/Packets/RoomUpdatePacket.cs:18]: This needs fixing in documentation to be more clear. Bleh. Thanks.

Comment (JarnoWesthof) [reply Archipelago.MultiClient.Net/Packets/RoomUpdatePacket.cs:18]: Agreed it also confused me at first

**Pull request merged**